mealie: 2.7.1 -> 2.8.0; python3Packages.ingredient-parser-nlp: init at 2.0.0

antonmosich opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Mealie has a new dependency in its latest update: ingredient-parser-nlp, which is added to nixpkgs in this PR as well.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

antonmosich committed:
nltk-data: add averaged_perceptron_tagger_eng

antonmosich committed:
python3Packages.ingredient-parser-nlp: init at 2.0.0

antonmosich committed:
mealie: 2.7.1 -> 2.8.0

crfpp has been replaced by ingredient-parser-nlp, so we no longer need
to have it around.
Changelog: https://github.com/mealie-recipes/mealie/releases/tag/v2.8.0

antonmosich commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 392158`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>python312Packages.ingredient-parser-nlp</li>
    <li>python312Packages.ingredient-parser-nlp.dist</li>
    <li>python313Packages.ingredient-parser-nlp</li>
    <li>python313Packages.ingredient-parser-nlp.dist</li>
  </ul>
</details>

litchipi reviewed (approved):
Seams alright, if tests are OK on your end let's merge this
Sorry I cannot test it on my end before quite a bit of time

antonmosich commented:
Both the package tests as well as `nixosTests.mealie` work without problems on my machine.

FliegendeWurst commented:
You need to resolve the merge conflict.

antonmosich commented:
Thanks for notifying me, I didn't notice the conflict

FliegendeWurst commented:
Hmm, it seems there is another merge conflict. This time around, I will try to review a bit quicker after you fix it :)

antonmosich commented:
Merge conflict should be resolved now. Don't worry, such conflicts are easily resolved.

FliegendeWurst commented:
The nixos test fails on aarch64-linux for ofborg with:

> server # [  564.833514] init_db[944]: /nix/store/cxhrrsf7spcgdkrxmlbfzc1bh46rzf7w-python3-3.12.9/bin/python3.12: can't open file '/nix/store/31k1nmm20ah5l2nmh9f6ars0x48fpsz9-mealie-2.8.0/lib/python3.12/site-packages/mealie/scripts/install_model.py': [Errno 2] No such file or directory

FliegendeWurst commented:
Looks like it was failing for the last few updates already, would still be good to have a look.

antonmosich committed:
nixos/mealie: update for mealie v2.8.0

mealie v2.8.0 no longer uses crfpp, but instead uses
ingredient_parser_nlp, which relies on nltk-data. If this environment
variable isn't available, mealie will just download the data instead.

FliegendeWurst commented on pkgs/development/python-modules/ingredient-parser-nlp/default.nix:
```diff
@@ -0,0 +1,58 @@
+  # Needed for tests
+  env.NLTK_DATA = nltk-data.averaged_perceptron_tagger_eng;
+
+  passthru.updateScript = nix-update-script { };
```
I don't think you need to specify this for Python packages, they all get an update script based on PyPI if I remember correctly.
LGTM otherwise, build is still running here.

antonmosich replied:
Thanks, I removed it

xanderio reviewed (approved):
Tests are now passing on aarch64. @FliegendeWurst I'm going to merge this later today or tomorrow morning.

ambroisie commented on pkgs/development/python-modules/ingredient-parser-nlp/default.nix:
```diff
@@ -0,0 +1,56 @@
+  ];
+
+  # Needed for tests
+  env.NLTK_DATA = nltk-data.averaged_perceptron_tagger_eng;
```
Nit: if it's only needed for tests, it probably should be set in `preCheck`:

```suggestion
  preCheck = ''
    export NLTK_DATA=${nltk-data.averaged_perceptron_tagger_eng}
  '';

ambroisie commented on pkgs/by-name/me/mealie/package.nix:
```diff
@@ -127,6 +107,9 @@ pythonpkgs.buildPythonApplication rec {
   nativeCheckInputs = with pythonpkgs; [ pytestCheckHook ];
 
+  # Needed for tests
+  env.NLTK_DATA = nltk-data.averaged_perceptron_tagger_eng;
```
Same note.

ambroisie reviewed (approved):
Small nit about `NLTK_DATA` in packages, will approve and merge regardless.

EDIT: unable to force push over the branch to fix my nits.

antonmosich commented:
Thanks, just addressed the review!
